src/lib/ImpToLLVM.hs

```diff
@@ -1096,10 +1077,48 @@ pointerType = const L.PointerType
 pointerType = L.PointerType
 #endif
 
+-- === Output stream ===
+
+getOutputStream :: LLVMBuilder m => m Operand
+getOutputStream = getDyvar OutStreamDyvar
+
 initializeOutputStream :: LLVMBuilder m => Operand -> m ()
 initializeOutputStream streamFD = do
   streamPtr <- emitExternCall fdopenFun [streamFD]
-  store outputStreamPtr streamPtr
+  setDyvar OutStreamDyvar streamPtr
+
+-- === Dynamically-scoped vars via thread-local pthreads vars ===
+
+dyvarDefs :: [L.Definition]
+dyvarDefs  = (:[]) $ L.GlobalDefinition $ L.globalVariableDefaults
+  { L.name = dynamicVarLName OutStreamDyvar
+  , L.type' = hostVoidp
+  , L.linkage = L.External
+  , L.initializer = Nothing }
```

Comment:
You could just make it a list comprehension over the `DynamicVar` enum, right?

Reply:
Nice. Good idea